core/input/input_event.h

```diff
@@ -319,15 +319,15 @@ class InputEventJoypadMotion : public InputEvent {
 class InputEventJoypadButton : public InputEvent {
 	GDCLASS(InputEventJoypadButton, InputEvent);
 
-	int button_index = 0;
+	JoyButton button_index = (JoyButton)0;
```

Comment:
Shouldn't it be  `JOY_BUTTON_INVALID` (-1) by default?

Reply:
Maybe. That would change behavior beyond the scope of this PR, so it can be done in another PR if desired.